posts/2023-03-31-managing-github-actions.md

```diff
@@ -0,0 +1,231 @@
+
+With one of our earliest pilots, a customer wanted to use a terraform action. It failed and I felt a bit embarrassed by the reason. We were missing unzip in our images.
+
+The cure? Go and add unzip to the Dockerfile, and hit publish on our builder repository. In 3 minutes the problem was solved.
+
+But GitHub Actions is also incredibly versatile and it means even if something is missing, we don't necessary have to publish a new image for you to continue your work. Just add a step to your workflow to install the missing package.
+
+```yaml
+- name: Add unzip
+  run: sudo apt-get install -qy unzip
+```
+
+With every customer pilot we've done, there's tended to be one or two packages like this that they expected to see. For another customer it was "libpq". As a rule, if something is available in the hosted runner, we'll strongly consider adding it to ours.
+
+## Lesson 2 - It’s not GitHub, it can’t be GitHub. It was GitHub
+
+Since actuated is a control-plane, a SaaS, a full-service - supported product, we are always asking first - is it us? Is it our code? Is it our infrastructure? Is it our network? Is it our hardware?
+
+If you open up the [GitHub status page](https://githubstatus.com), you'll notice an outage almost every week - at times on consecutive days, or every few days on GitHub Actions or a service that affects them indirectly - like the package registry, Pages or Pull Requests.
+
+![Outage this week](/images/2023-03-lessons-learned/outage.png)
+> The second outage this week that unfortunately affected actuated customers.
+
+I'm not bashing on GitHub here, we're paying a high R&D cost to build on their platform. We want them to do well.
+
+But this is getting embarrassing. On a recent call, a customer told us: "it's not your solution, it looks great for us, it's the reliability of GitHub, we're worried about adopting it"
+
+What can you say to that? I can't tell them that their concerns are misplaced, because they're not.
+
+I reached out to Martin Woodward - Director of DevRel at GitHub. He told me that "leadership are taking this very seriously. We're doing better than we were 12 months ago."
+
+GitHub is too big to fail. Let's hope they smooth out these bumps.
+
+## Lesson 3 - Webhooks can be unreliable
+
+There's no good API to collect this historical data at the moment but we do have an open-source tool ([self-actuated/actions-usage](https://github.com/self-actuated/actions-usage)) we give to customers to get a summary of their builds before they start out with us.
+
+So we mirror a summary of job events from GitHub into our database, so that we can show customers trends in behaviour, and identify hot-spots on specific repos - long build times, or spikes in failure rates.
+
+![Insights chart](https://pbs.twimg.com/media/FqnJ8rLXgAEnJDZ?format=png&name=large)
```

Comment:
Maybe can use a newer picture of the insights with the updated tool tip that displays more information.

Reply:
I was thinking that too, but this one had the best data on the chart.

One thing we need to do separately is to add a section into the docs about the dashboard's insights tab.